plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -0,0 +1,420 @@
+
+        assertQueryFails(format("CALL iceberg.system.register_table (CURRENT_SCHEMA, '%s')", tableName),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA)",
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table ()",
+                ".*'SCHEMA_NAME' is missing.*");
+
+        assertQueryFails("CALL iceberg.system.register_table (null, null, null)",
+                ".*schema_name cannot be null or empty.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, null, null)",
+                ".*table_name cannot be null or empty.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, '" + tableName + "', null)",
+                ".*table_location cannot be null or empty.*");
+
+        assertQueryFails("CALL iceberg.system.register_table ('', '" + tableName + "', '" + tableLocation + "')",
+                ".*schema_name cannot be null or empty.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, '', '" + tableLocation + "')",
+                ".*table_name cannot be null or empty.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, '" + tableName + "', '')",
+                ".*table_location cannot be null or empty.*");
+        assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "', '')",
+                ".*metadata_file_name cannot be empty when provided as an argument.*");
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidMetadataFileName()
+    {
+        String tableName = "test_register_table_with_invalid_metadata_file_name_" + randomTableSuffix();
+        String tableLocation = "/test/iceberg/hive";
+
+        String[] invalidMetadataFileNames = {
+                "/",
+                "../",
+                "../../",
+                "../../somefile.metadata.json",
+        };
+
+        for (String invalidMetadataFileName : invalidMetadataFileNames) {
+            assertQueryFails("CALL iceberg.system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "', '" + invalidMetadataFileName + "')",
+                    ".*is not a valid metadata file.*");
+        }
+    }
+
+    private String getTableLocation(String tableName)
+    {
+        return (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*/[^/]*$', '') FROM " + tableName);
+    }
+
+    private void dropTableFromMetastore(String tableName)
+    {
+        metastore.dropTable(getSession().getSchema().orElseThrow(), tableName, false);
+        tableNotExist(tableName);
+    }
+
+    private void tableNotExist(String tableName)
+    {
+        assertThat(metastore.getTable(getSession().getSchema().orElseThrow(), tableName)).as("Table in metastore should be dropped").isEmpty();
```

Comment:
nit: This assertion can be moved directly into the method `dropTableFromMetastore`